src/components/Dashboard.tsx

```diff
@@ -1,46 +1,90 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react'
+import logo from '../../public/images/logo.PNG'
 import Image from 'next/image';
-import { UserData, ComponentProps } from '@/types/types';
-import { getTodaysDate } from '@/utils';
+import { getTodaysDate, getZodiacSign } from '@/utils';
 import Navbar from '../components/Navbar';
 import { insights } from '@/mockdata';
 import Router from "next/router";
-import Logo from '@/components/logo';
+import { Passage } from '@passageidentity/passage-js';
+import { PassageUserInfo } from '@passageidentity/passage-elements/passage-user';
+import dotenv from 'dotenv';
+import { AuthProps } from '@/types/types';
 
+const getCurrentUser = async () => {
+  dotenv.config();
+  const passageAppId = process.env.NEXT_PUBLIC_PASSAGE_APP_ID;
 
-type DashboardProps = ComponentProps & {
-  logOut: () => void
-};
+  if (!passageAppId) {
+    throw new Error('NEXT_PUBLIC_PASSAGE_APP_ID is not defined in the environment.');
+  }
 
+  const passage = new Passage(passageAppId);
+  const user = passage.getCurrentUser();
+  const userInfo = await user.userInfo();
+  return userInfo
+}
 
-export default function Dashboard({ isAuthorized, data, logOut}: DashboardProps) {
-  const [user, setUser] = useState<UserData | null>(null)
-  const [userInsights, setUserInsights] = useState(insights)
-
+export default function Dashboard({ isAuthorized, userID }: AuthProps) {
+  const [user, setUser] = useState<PassageUserInfo | undefined>(undefined);
+  // const [userInsights, setUserInsights] = useState(insights)
 
   useEffect(() => {
     if (!isAuthorized) {
       Router.push("/");
-    }      
-  
-    if (data) setUser(data[0])
-    //api call needs to be made to get horoscope
+    } else {
+      getCurrentUser()
+        .then((data) => {
+          setUser(data)
+          checkForUser(userID)
+        })
+        .catch((error) => {
+          console.error('Error fetching user:', error);
+        });
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [isAuthorized]);
-  
+
+  const checkForUser = async (userID: string | number) => {
+    const res = await fetch('/api/getUsers')
+    .then((res) => res.json())
+    .then((allIDs) => {
+      if (!allIDs.includes(userID) && user) {
+        addNewUser(user)
+      }
+    })
+  }
+
+  const addNewUser = async (user: PassageUserInfo ) => {
+    const payload = {
+      userID,
+      name: user.user_metadata?.name,
+      email: user.email,
+      birthday: user.user_metadata?.birthday,
+      sign: getZodiacSign(user.user_metadata?.birthday),
+    }
+    const res = await fetch('/api/addUser', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(payload),
+    }).then((res) => res.json())
+  };
```

Comment:
I think this should be on the backend, we can't accept email in a payload from the client because an evil user could modify their client code to send whatever they want. We should use the passage library to grab all of these user details from the JWT on the backend (so we know that it hasn't been tampered with)

Reply:
For sure! Just to clarify, this function is to add the user data from passage to our table on supabase; we can remove the email altogether, I only added it since it was a part of our original schema to add to our supabase table but it's not really necessary. For the last part, "use the passage library to grab all of these user details from the JWT on the backend", I have no idea how to do that lol but I am down to check it out and implement it that way!

Reply:
We can actually just upsert (If a record with a specified primary key (the passage userID) does not exist in the table, Supabase will insert a new record with that primary key and the specified data. This way we don't have to get all the users and manually check) a user in the DB in `getServerSideProps` since we already have to authenticate the user and have access to the information provided by passage. If we are calling a function to grab all the users on the front end, we will be exposing everyone's unique and secret identifier to the client, which is potentially accessible by anyone and then our DB is compromised

Reply:
Oh okay gotcha! That sounds way easier and safer than the route that I took lol. I am down to refactor!